File: Algorithms/5-2-2018/py/algos.py
# algos py 


def un_zip(obj):
    return [list(obj.keys()),list(obj.values())]

def countChars(obj):
    new_obj = {} 
    for key in obj: 
        if key in new_obj:
            new_obj[key] += 1 
        else :
            new_obj[key] = 1
    return new_obj

File: Algorithms/5-2-2018/js/algos.js


var methods = {}


var unZip = function(obj){
    return [
        Object.keys(obj),
        Object.keys(obj).map(x => obj[x])
    ]
}

var countChars = function(str){
    return str.split('').reduce((acc,item) => {
        if(!acc.hasOwnProperty(item))
            acc[item] = 1
        else
            acc[item] += 1
        return acc 
    },{})
}

methods.unZip = unZip
methods.countChars = countChars

module.exports = methods 